contributing/topics/guide-breaking-changes.md

```diff
@@ -1,8 +1,12 @@
 # Guide: Breaking Changes and Deprecations
 
-To keep up with and accommodate the changing pace of Azure, the provider needs to be able to introduce and handle breaking changes. We can safely introduce breaking changes into the provider using the major release feature flag, for the next major release that would the `features.FivePointOhBeta` flag which is available in the provider today.
+To keep up with and accommodate the changing pace of Azure, the provider needs to be able to gracefully introduce and handle breaking changes. A "breaking change" within the provider is considered to be anything that requires an end user to modify previously valid terraform configuration after a provider upgrade to either deploy new resources or to maintain existing deployments. Even if a change does not affect the user's current deployment, it is still considered a breaking change if it requires the user to modify their configuration to deploy new resources. 
 
-This guide includes several topics on how to do common deprecations and breaking changes in the provider using this feature flag, as well as additional guidance on how to deal with changing default values in the Azure API.
+The `azurerm` provider attempts to be as "surface stable" as possible during minor and patch releases meaning breaking changes are typically only made during major releases, however exceptions are sometimes made for minor releases when the breaking change is deemed necessary or is unavoidable. Terraform users rely on the stability of Terraform providers as not only can configuration changes be costly to make, test, and deploy they can also affect downstream tooling such as modules. Even as part of a major release, breaking changes that are overly large or have little benefit can delay users upgrading to the next major version.
+
+Generally we can safely introduce breaking changes into the provider for the major release using a feature flag. For the next major release that would be the `features.FivePointOhBeta()` flag which is available in the provider today. This guide includes several topics on how to do common deprecations and breaking changes in the provider using this feature flag, as well as additional guidance on how to deal with changing default values in the Azure API. 
```

Comment:
thanks for updating this guide.

In a recent code review, I ran acceptance tests by forcing the provider to be on the latest version.

It's worth updating the guide to call out running tests that exercise the current code path (which is already documented) and the code path for a future release (5.0).

Reply:
While that can be useful to check the flag works there is an overheard to doing so and sometimes changes which don't appear to affect 5.0 might slip in with a PR that doesn't obviously touch this

when we get closer to 5.0 we will start testing the 5.0 flag and looking for new test failures over main so i think its fine to not call it out and leave it for automated testing